### ISSUE TITLE
Limit XP gains for companions

### ISSUE THREAD

**Issue by sirzooro** [TLDRMissions/TLDRMissions#525]
**TLDR Version of your idea**
Prefer high level companions for missions with lots of XP to avoid leveling low level companions as much as possible.

**Slightly longer version**
Mission level also depends on average (or median) level of companions. Because of this it is better to keep companions on low levels as long as possible. This would make missions easier for troops, and TLDR would spend less time finding winning team. Initially I thought about simple limit for mission XP (e.g. up to 1000 XP), however some missions which I want to sent have 1200 XP (pet charms) or 1500 (resource bags). This needs other approach which would take into account both mission XP and companion level. For example for high-XP missions (1000+ XP) TLDR should start simulations with high-level companions first as they need more XP to level or do not level at all when they are at 60. Probably some way to assign weight to missions would be good here, e.g. `min((mission_XP / 1500), 1)`.

Some extra thought is also needed to determine how it should play together with existing option to send higher or lower level followers first. Now I have it set to prefer lower ones, to keep high level ones available as long as possible for difficult missions which have lower ranks assigned on 1st TLDR tab. However this setting also promotes leveling of low-level followers, so I would like to limit this aspect as much as possible with this new proposed functionality.

**Comment by teelolws** [comment]
Check the drop down menus next to the "Special treatment" option.
